streaming/src/test/scala/org/apache/spark/streaming/CheckpointSuite.scala

```diff
@@ -715,6 +736,35 @@ class CheckpointSuite extends TestSuiteBase with DStreamCheckpointTester {
     }
   }
 
+  test("DStreamCheckpointData.restore invoking times") {
+    var clock: ManualClock = null
+    val outputBuffer = new ArrayBuffer[Seq[Seq[Int]]] with SynchronizedBuffer[Seq[Seq[Int]]]
```

Comment:
You can remove `outputBuffer` and change `val outputStream = new TestOutputStreamWithPartitions(mappedDStream, outputBuffer)` to `val outputStream = new TestOutputStreamWithPartitions(mappedDStream)` since the output buffer is not necessary now.